### ISSUE TITLE
[BUG] Callbacks executed after patience delay

### ISSUE THREAD

**Issue by vince2848** [GRAAL-Research/poutyne#109]
**Bug description**
The EarlyStop callback seems to execute one more epoch before the training stops.
The same thing happens with ReduceLROnPlateau.

**To Reproduce**
```
model = Model(net, "adam", "cross_entropy", batch_metrics=['accuracy'])
callbacks = [pt.EarlyStopping(patience=10, monitor = 'val_acc', mode='max')]
model.fit_generator(train_dataloader, val_dataloader, epochs=100, callbacks=callbacks)
```

**Expected behavior**
The training should stops after the patience is over.

**Screenshots**
This one was executed with patience = 5. It stopped after the last displayed epoch.
![image](https://user-images.githubusercontent.com/2975698/114877376-3a7c1900-9dcd-11eb-8c83-d583fd17d7c0.png)

This one was executed with EarlyStop patience = 5 and ReduceLROnPlateau patience = 2.
![image](https://user-images.githubusercontent.com/2975698/114878274-140aad80-9dce-11eb-989f-ac4eba992964.png)

**Desktop (please complete the following information):**
Google colab

**Comment by freud14** [comment]
Hi,

Thank you for the issue. There was indeed a bug in the EarlyStopping callback. I just pushed a fix in the dev branch of the project. It will be released in the next version.

As for the ReduceLROnPlateau callback, I just use PyTorch's class for this and it is working as intended according to the [documentation](https://pytorch.org/docs/stable/optim.html#torch.optim.lr_scheduler.ReduceLROnPlateau):
> patience (int) – Number of epochs with no improvement after which learning rate will be reduced. **For example, if patience = 2, then we will ignore the first 2 epochs with no improvement, and will only decrease the LR after the 3rd epoch if the loss still hasn’t improved then.** Default: 10.

Emphasis is mine.

Let me know if this reply satisfies you.

Thank you.

Frédérik

**Comment by freud14** [comment]
Hi,

I just made a new release. You should be able to get the fix using "pip install poutyne".

Thank you.

Frédérik
